nixos/modules/services/networking/pihole-ftl.nix

```diff
@@ -0,0 +1,107 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+
+with lib;
+
+let
+
+  cfg = config.services.pihole-ftl;
+  stateDir = "/etc/pihole";
+  confFile = "${stateDir}/pihole-FTL.conf";
+  logFile = "/var/log/pihole-FTL.log";
+  confText = pkgs.writeText "pihole-FTL.conf" cfg.config;
+
+in
+{
+
+  options.services.pihole-ftl = {
+    enable = mkEnableOption "Pi-hole FTL";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Open ports in the firewall for Pi-hole FTL.";
+    };
+
+    config = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Configuration file content for Pi-hole FTL.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pihole-ftl;
+      defaultText = "pkgs.pihole-ftl";
+      description = "Pi-hole FTL package to use.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = !config.services.dnsmasq.enable;
+        message = "pihole-ftl conflicts with dnsmasq";
+      }
+    ];
+
+    systemd.tmpfiles.rules = [
+      "d ${stateDir} 0700 pihole pihole - -"
+      "L+ ${confFile} - - - - ${confText}"
+      "f ${logFile} 0700 pihole pihole - -"
```

Comment:
Will this grow indefinitely until the system it runs on comes to a crash from lack of disk space?

Reply:
It will keep 24h I think https://docs.pi-hole.net/ftldns/configfile/#maxlogage

Reply:
I saw a number of mentions about log rotation, so maybe the old files need to be deleted. NixOS has a module for that, if needed.